kernel/aster-nix/src/util/mod.rs

```diff
@@ -106,39 +105,32 @@ pub fn write_val_to_user<T: Pod>(dest: Vaddr, val: &T) -> Result<()> {
 /// the `do_strncpy_from_user` function in Linux kernel.
 /// The original Linux implementation can be found at:
 /// <https://elixir.bootlin.com/linux/v6.0.9/source/lib/strncpy_from_user.c#L28>
-pub fn read_cstring_from_user(addr: Vaddr, max_len: usize) -> Result<CString> {
-    let current = current!();
-    let vmar = current.root_vmar();
-    read_cstring_from_vmar(vmar, addr, max_len)
-}
-
-/// Read CString from `vmar`. If possible, use `read_cstring_from_user` instead.
-pub fn read_cstring_from_vmar(vmar: &Vmar<Full>, addr: Vaddr, max_len: usize) -> Result<CString> {
+pub fn read_cstring_from_user(vaddr: Vaddr, max_len: usize) -> Result<CString> {
```

Comment:
Excellent job here! All syscalls who invoke `read_cstring_from_user()` will gain great performance improvement through this revision. (Most are fs-related such as `sys_openat`, `sys_stat`, `sys_truncate`, `sys_mount`).
Take our benchmarks in CI as examples, lmbench-open (~2.9ms -> ~1.6ms), lmbench-stat (~2.3ms -> ~1.0ms).